### ISSUE TITLE
enabling MICROPY_CPYTHON_COMPAT fails to build for nrf

### ISSUE THREAD

**Issue by jerryneedell** [adafruit/circuitpython#1389]
The nrf52 builds fail to build if MICROPY_CPYTHON_COMPAT is enabled

```
jerryneedell@Ubuntu-Macmini:~/circuitpython_jerryn/ports/nrf$ make BOARD=particle_argon 
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
QSTR updated
FREEZE freeze
LINK build-particle_argon/firmware.elf
build-particle_argon/py/modio.o:(.rodata.mp_module_io_globals_table+0x1c): undefined reference to `mp_type_textio'
collect2: error: ld returned 1 exit status
Makefile:345: recipe for target 'build-particle_argon/firmware.elf' failed
make: *** [build-particle_argon/firmware.elf] Error 1
```

**Comment by jerryneedell** [comment]
to fix this and enable CPYTHON_COMPAT and a few other items that are different from the ATMEL SAMD51 builds
```
jerryneedell@Ubuntu-Macmini:~/circuitpython_jerryn/ports/nrf$ git diff
diff --git a/ports/nrf/mpconfigport.h b/ports/nrf/mpconfigport.h
index 86f5af20a..8b1c27367 100644
--- a/ports/nrf/mpconfigport.h
+++ b/ports/nrf/mpconfigport.h
@@ -74,16 +74,16 @@
 #define mp_builtin_open_obj mp_vfs_open_obj
 #endif
 
-#define MICROPY_CPYTHON_COMPAT                   (0)
+#define MICROPY_CPYTHON_COMPAT                   (1)
 #define MICROPY_STREAMS_NON_BLOCK                (1)
 #define MICROPY_MODULE_WEAK_LINKS                (1)
 #define MICROPY_CAN_OVERRIDE_BUILTINS            (1)
 #define MICROPY_USE_INTERNAL_ERRNO               (0)
 #define MICROPY_PY_FUNCTION_ATTRS                (1)
 #define MICROPY_PY_BUILTINS_STR_UNICODE          (1)
-#define MICROPY_PY_BUILTINS_STR_CENTER           (0)
-#define MICROPY_PY_BUILTINS_STR_PARTITION        (0)
-#define MICROPY_PY_BUILTINS_STR_SPLITLINES       (0)
+#define MICROPY_PY_BUILTINS_STR_CENTER           (1)
+#define MICROPY_PY_BUILTINS_STR_PARTITION        (1)
+#define MICROPY_PY_BUILTINS_STR_SPLITLINES       (1)
 #define MICROPY_PY_BUILTINS_MEMORYVIEW           (1)
 #define MICROPY_PY_BUILTINS_FROZENSET            (1)
 #define MICROPY_PY_BUILTINS_EXECFILE             (0)
@@ -155,6 +155,8 @@ typedef long mp_off_t;
 
 #define MP_PLAT_PRINT_STRN(str, len) mp_hal_stdout_tx_strn_cooked(str, len)
 #define mp_type_fileio mp_type_vfs_fat_fileio
+#define mp_type_textio mp_type_vfs_fat_textio
+
 
 // extra built in modules to add to the list of known ones
```
the build succeeds with these changes.
```Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.

Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 4.0.0-alpha-1529-g2b739e1eb-dirty on 2018-12-09; Particle Argon with nRF52840
>>> dir(str)
['__class__', '__name__', 'center', 'count', 'encode', 'endswith', 'find', 'format', 'index', 'isalpha', 'isdigit', 'islower', 'isspace', 'isupper', 'join', 'lower', 'lstrip', 'partition', 'replace', 'rfind', 'rindex', 'rpartition', 'rsplit', 'rstrip', 'split', 'splitlines', 'startswith', 'strip', 'upper']
>>> dir(bytes)
['__class__', '__name__', 'center', 'count', 'decode', 'endswith', 'find', 'format', 'index', 'isalpha', 'isdigit', 'islower', 'isspace', 'isupper', 'join', 'lower', 'lstrip', 'partition', 'replace', 'rfind', 'rindex', 'rpartition', 'rsplit', 'rstrip', 'split', 'splitlines', 'startswith', 'strip', 'upper']
>>> 

```

**Comment by jerryneedell** [comment]
Should I create a PR for this or wait for a complete review of mpconfigport.h for the nrf builds?

**Comment by ladyada** [comment]
@jerryneedell heya is this issue resolved?

**Comment by jerryneedell** [comment]
yes - fixed by #1395
